kubernetes/structures.go

```diff
@@ -77,6 +77,17 @@ func expandStringMap(m map[string]interface{}) map[string]string {
 	return result
 }
 
+func expandBase64MapToByteMap(m map[string]interface{}) map[string][]byte {
```

Comment:
As a personal preference I like to have these expanders directly return `map[string]interface{}` just because that's what being used from there on and it reads nicer in the calling function when there's no type assertion.

There is no consistency of this approach throughout the code though.

Reply:
> There is no consistency of this approach throughout the code though.

 Yeah. Also, the naming of the various expander/flatteners functions left me perplex there.

Reply:
> As a personal preference I like to have these expanders directly return `map[string]interface{}` just because that's what being used from there on and it reads nicer in the calling function when there's no type assertion.

Turns out none of the already existing `expandStringMap()`, `expandStringMapToByteMap()` and `expandStringSlice()` return `map[string]interface{}` and I've modeled `expandBase64MapToByteMap` on `expandStringMapToByteMap()`.

In fact, wouldn't this render `expandStringMap()` a no-op?
Is this function even needed?